pkg/deploy/generator/resources.go

```diff
@@ -151,7 +151,14 @@ func (g *generator) proxyVmss() *arm.Resource {
 		)
 	}
 
-	trailer := base64.StdEncoding.EncodeToString([]byte(`yum -y update -x WALinuxAgent
+	trailer := base64.StdEncoding.EncodeToString([]byte(`
+	
+if [$LOCATION == southeastasia] then
```

Comment:
the bash will surely not run because there are syntax errors
also the PR will fail unit tests because you didn't run make generate
does a vanilla Azure RHEL 8 VM have dig? might be easier to parse: `dig +short australiaeast-cds.australiaeast.cloudapp.azure.com`

Reply:
updated